src/coreclr/vm/methodtable.cpp

```diff
@@ -704,8 +670,17 @@ void MethodTable::AllocateAuxiliaryData(LoaderAllocator *pAllocator, Module *pLo
     pMTAuxiliaryData = (MethodTableAuxiliaryData *)(pAuxiliaryDataRegion + prependedAllocationSpace);
 
     pMTAuxiliaryData->SetLoaderModule(pLoaderModule);
-    pMTAuxiliaryData->SetOffsetToNonVirtualSlots(hasGenericStatics ? -(int16_t)sizeof(GenericsStaticsInfo) : 0);
+    pMTAuxiliaryData->SetOffsetToNonVirtualSlots(-sizeofStaticsStructure);
```

Comment:
Please add a comment. That `-` is a bit subtle.

Reply:
LOL. I can see missing that one easily. I'll add

`// NOTE: There is a - sign here making it so that the offset points to BEFORE the MethodTableAuxiliaryData`